docs/protocol-p2p-network/initial-block-download.md

```diff
@@ -19,3 +19,125 @@ best block chain has a block header time more than 24 hours in the past.
+sent in order because each block header references the header hash of
+the preceding block. That means the IBD node can't fully validate a
+block until its parent block has been received. Blocks that can't be
+validated because their parents haven't been received are called [orphan
+blocks](orphan-blocks.md).
+
+Upon receipt of the `getdata` message, the sync node replies with each
+of the blocks requested. Each block is put into serialized block format
+and sent in a separate `block` message. The first `block` message sent
+(for block 1) is illustrated below.
+
+![First Block Message Sent During IBD](/img/protocol-p2p-network/en-ibd-block.svg)
+
+The IBD node downloads each block, validates it, and then requests the
+next block it hasn't requested yet, maintaining a queue of up to 128
+blocks to download. When it has requested every block for which it has
+an inventory, it sends another `getblocks` message to the sync node
+requesting the inventories of up to 500 more blocks.  This second
+`getblocks` message contains multiple header hashes as illustrated
+below:
+
+![Second GetBlocks Message Sent During IBD](/img/protocol-p2p-network/en-ibd-getblocks2.svg)
+
+Upon receipt of the second `getblocks` message, the sync node searches
+its local best block chain for a block that matches one of the header
+hashes in the message, trying each hash in the order they were received.
+If it finds a matching hash, it replies with 500 block inventories
+starting with the next block from that point. But if there is no
+matching hash (besides the stopping hash), it assumes the only block the
+two nodes have in common is block 0 and so it sends an `inv` starting with
+block 1 (the same `inv` message seen several illustrations above).
+
+This repeated search allows the sync node to send useful inventories even if
+the IBD node's local block chain forked from the sync node's local block
+chain. This fork detection becomes increasingly useful the closer the
+IBD node gets to the tip of the block chain.
+
+When the IBD node receives the second `inv` message, it will request
+those blocks using `getdata` messages.  The sync node will respond with
+`block` messages.  Then the IBD node will request more inventories with
+another `getblocks` message---and the cycle will repeat until the IBD
+node is synced to the tip of the block chain.  At that point, the node
+will accept blocks sent through the regular
+[block broadcasting](block-broadcasting.md) mechanism.
+
+### Blocks-First Advantages & Disadvantages
+
+The primary advantage of blocks-first IBD is its simplicity. The primary
+disadvantage is that the IBD node relies on a single sync node for all
+of its downloading. This has several implications:
+
+* **Speed Limits:** All requests are made to the sync node, so if the
+  sync node has limited upload bandwidth, the IBD node will have slow
+  download speeds.  Note: if the sync node goes offline, dcrd
```

Comment:
Note feels awkward here...Suggest:

Note that if the sync node goes offline, dcrd
  will continue downloading from another node, but it will still only
  download from a single sync node at a time.